### PR TITLE
cloud protocol v5

### PR DESCRIPTION
**Pull request by Eabramov84**

### CODE REVIEW

**Review by MykolaSolyanko — APPROVED** [review]

File: amqphandler/amqphandler.go [hunk 1]
```diff
@@ -596,154 +596,29 @@ func (handler *AmqpHandler) runReceiver(deliveryChannel <-chan amqp.Delivery, pa
...
-) {
-	var secret cloudprotocol.UnitSecret
+			log.Infof("AMQP receive message: %s", incomingMsg.Header.MessageType)
 
-	if len(encodedNotification.UnitSecureData) > 0 {
-		if err := handler.decodeData(encodedNotification.UnitSecureData, &secret); err != nil {
-			return nil, aoserrors.Wrap(err)
-		}
+			if err := handler.decodeData(rawData, decodedData); err != nil {
+				log.Errorf("Can't decode incoming message %s", err)
```
Comment (al1img) [inline amqphandler/amqphandler.go:615]: : %v

File: amqphandler/amqphandler_test.go [hunk 2]
```diff
@@ -169,8 +166,13 @@ func cleanup() {
 	}
 }
 
-func sendMessage(message interface{}) error {
```
Comment (al1img) [inline amqphandler/amqphandler_test.go:172]: What the reason to change this? You have to duplicate send message and expected message in test data.

File: monitorcontroller/monitorcontroller.go [hunk 3]
```diff
@@ -142,13 +142,12 @@ func (monitor *MonitorController) processQueue(ctx context.Context) {
...
-
-				monitoringData, monitor.monitoringQueue = monitor.monitoringQueue[0], monitor.monitoringQueue[1:]
-
 				if err := monitor.monitoringSender.SendMonitoringData(
-					monitoringData); err != nil && !errors.Is(err, amqphandler.ErrNotConnected) {
+					cloudprotocol.Monitoring{Nodes: monitor.monitoringQueue}); err != nil &&
+					!errors.Is(err, amqphandler.ErrNotConnected) {
 					log.Errorf("Can't send monitoring data: %v", err)
+				} else {
+					monitor.monitoringQueue = make([]cloudprotocol.NodeMonitoringData, 0, monitor.monitoringQueueSize)
```
Comment (al1img) [inline monitorcontroller/monitorcontroller.go:150]: Should we clear queue in any case?

Comment (Eabramov84) [reply monitorcontroller/monitorcontroller.go:150]: from my point of view we have to collect data in case of failure case

**Review by al1img — APPROVED** [review]

File: amqphandler/amqphandler_test.go (hunk 2 above)
Comment (Eabramov84) [reply amqphandler/amqphandler_test.go:172]: resolves

**Pull request merged**